flask/src/main/java/net/corda/gradle/flask/FlaskJarTask.java

```diff
@@ -0,0 +1,249 @@
+package net.corda.gradle.flask;
+
+import lombok.RequiredArgsConstructor;
+import lombok.SneakyThrows;
+import net.corda.flask.common.Flask;
+import org.gradle.api.Action;
+import org.gradle.api.NamedDomainObjectCollection;
+import org.gradle.api.NamedDomainObjectContainer;
+import org.gradle.api.internal.file.CopyActionProcessingStreamAction;
+import org.gradle.api.internal.file.copy.CopyAction;
+import org.gradle.api.internal.file.copy.CopyActionProcessingStream;
+import org.gradle.api.internal.file.copy.FileCopyDetailsInternal;
+import org.gradle.api.model.ObjectFactory;
+import org.gradle.api.plugins.BasePluginConvention;
+import org.gradle.api.provider.ListProperty;
+import org.gradle.api.provider.Property;
+import org.gradle.api.provider.Provider;
+import org.gradle.api.tasks.Input;
+import org.gradle.api.tasks.Nested;
+import org.gradle.api.tasks.WorkResult;
+import org.gradle.api.tasks.bundling.AbstractArchiveTask;
+
+import javax.annotation.Nonnull;
+import javax.inject.Inject;
+import java.io.*;
+import java.nio.file.attribute.FileTime;
+import java.security.MessageDigest;
+import java.util.*;
+import java.util.function.Supplier;
+import java.util.jar.Attributes;
+import java.util.jar.JarFile;
+import java.util.jar.Manifest;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+import java.util.zip.ZipOutputStream;
+
+
+public class FlaskJarTask extends AbstractArchiveTask {
+
+    private final Property<String> launcherClassName;
+
+    @Input
+    public Property<String> getLauncherClassName() {
+        return launcherClassName;
+    }
+
+    private final Property<String> mainClassName;
+
+    @Input
+    public Property<String> getMainClassName() {
+        return mainClassName;
+    }
+
+    private final ListProperty<String> jvmArgs;
+
+    @Input
+    public ListProperty<String> getJvmArgs() {
+        return jvmArgs;
+    }
+
+    public void includeLibraries(Object files) {
```

Comment:
This can become
```java
public void includeLibraries(Object... files) {
   //
}
```
because `copySpec.from()` accepts `Object...` too. (Java is more sensible than Kotlin about assigning `varargs` parameters.)